src/Core/Managed/Shared/Extensibility/Implementation/SimpleMetricStatisticsAggregator.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// Represents aggregator for a single time series of a given metric.
+    /// </summary>
+    internal class SimpleMetricStatisticsAggregator
```

Comment:
Ok, so how about calling them `Metric` and `SimpleMetricStatisticAggregator` then?